Framework/HistogramData/src/HistogramItem.cpp

```diff
@@ -0,0 +1,106 @@
+#include "MantidHistogramData/HistogramItem.h"
+#include "MantidHistogramData/Histogram.h"
+#include <algorithm>
+#include <utility>
+
+using Mantid::HistogramData::HistogramItem;
+using Mantid::HistogramData::Histogram;
+using Mantid::HistogramData::BinEdges;
+using Mantid::HistogramData::Points;
+
+HistogramItem::HistogramItem(const Histogram &histogram, const size_t index)
+    : m_histogram(histogram), m_index(index) {}
+
+double HistogramItem::counts() const {
+  const auto &yMode = histogramRef().yMode();
+  if (yMode == Histogram::YMode::Counts) {
+    return histogramRef().counts()[m_index];
+  } else {
+    const Frequencies frequency{histogramRef().frequencies()[m_index]};
+    return Counts(frequency, binEdges())[0];
+  }
+}
+
+double HistogramItem::countVariance() const {
+  return histogramRef().countVariances()[m_index];
+}
+
+double HistogramItem::countStandardDeviation() const {
+  return histogramRef().countStandardDeviations()[m_index];
+}
+
+double HistogramItem::frequency() const {
+  const auto &yMode = histogramRef().yMode();
+  if (yMode == Histogram::YMode::Frequencies) {
+    return histogramRef().frequencies()[m_index];
+  } else {
+    const Counts counts{histogramRef().counts()[m_index]};
+    return Frequencies(counts, binEdges())[0];
+  }
+}
+
+double HistogramItem::frequencyVariance() const {
+  return histogramRef().frequencyVariances()[m_index];
+}
+
+double HistogramItem::frequencyStandardDeviation() const {
+  return histogramRef().frequencyStandardDeviations()[m_index];
+}
+
+double HistogramItem::center() const { return point()[0]; }
+
+double HistogramItem::width() const {
+  const auto &edges = binEdges();
+  const auto &lower = edges[0];
+  const auto &upper = edges[1];
+  return upper - lower;
+}
+
+const BinEdges HistogramItem::binEdges() const {
```

Comment:
I hesitated for a second here, the return type kind of works, but in the end I think it is probably too confusing. Additionally, creating `BinEdges` involves two memory allocation (`shared_ptr` and `vector`) so it should be too expensive for this purpose.

Same goes for using `Points` below.